src/clj/game/cards/resources.clj

```diff
@@ -2009,6 +2037,35 @@
     :leave-play (req (swap! state update-in [:corp] dissoc :cannot-win-on-points)
                      (gain-agenda-point state :corp 0))}
 
+   "The Class Act"
+   {:effect (req true)
+    :end-turn {:async true
+               :msg "draw 4 cards"
+               :effect (effect (draw eid 4 nil))}
+    :events {:pre-runner-draw ; TODO: ensure this doesnt trigger if runner draws, uh, 0
+             {:msg "draw 1 additional card"
+              ;; The req catches draw events that happened before The Class Act was installed
+              :req (req (first-event? state :runner :pre-runner-draw))
+              :once :per-turn
+              :once-key :class-act-extra-draw ; TODO: is this even needed? i think it is because both abilities must be :once but check
```

Comment:
Yes, this is needed, because you have two different `:once :per-turn` effects that both need to fire a single time.